src/OptionsCompounderAave2.sol

```diff
@@ -128,38 +96,38 @@ contract OptionsCompounder is FlashLoanReceiverBase, Ownable {
         ) {
             revert OptionsCompounder__TooMuchAssetsLoaned();
         }
+        /* Later the gain can be local variable */
         gain = exerciseOptionAndReturnDebt(
             assets[0],
             amounts[0],
             premiums[0],
             params
         );
+        flashloanFinished = true;
         return true;
     }
 
     /**
      * @dev function initiate flashloan in order to exercise option tokens and compound rewards
      * in underlying tokens to want token
      */
-    function harvestOTokens(
-        uint256 amount,
-        address option
-    ) public onlyStrategy {
-        if (false == optionToken.isOption(option)) {
+    // TODO: Add access control
+    function harvestOTokens(uint256 amount, address option) external {
```

Comment:
this interface will change "option" will be called "exerciseContract" or smth @lookeey what did u land on